game_options.cpp

```diff
@@ -110,12 +131,27 @@ void test_game_options()
   }
   // Two random game options differ in their key-action maps especially
   {
+    const int rng_seed = 111;
     const game_options a = get_random_game_options(rng_seed);
     const game_options b = get_random_game_options(rng_seed + 1);
     assert(a.get_kam_1() != b.get_kam_1());
     assert(a.get_kam_2() != b.get_kam_2());
   }
   #endif // FIX_ISSUE_303
 
+  #ifdef FIX_ISSUE_353
+  {
+    // Random game options should not draw keys already used by another player
+    const int rng_seed = 271; // by default this will assign A to both players
+    const game_options rgo = get_random_game_options(rng_seed);
+    key_action_map map_1 = rgo.get_kam_1();
+    key_action_map map_2 = rgo.get_kam_2();
+    bool a = map_1.has_key(sf::Keyboard::Key::A);
+    bool b = map_2.has_key(sf::Keyboard::Key::A);
+    // Key A is in KAM1 or KAM2 but not both
+    assert(a != b);

```

Comment:
This is a scary assert, chance 1 in 26^2 it will be triggered. This will happen one day. Let's keep it as-is, so a junior can 'debug' the problem when it happens :-)

Reply:
I set `rng_seed = 271` because I know that drawing random keys with this seed assigns key A to both players 😉 
I just ran the program three times, the two key action maps were assigned the same values in all three iterations.
Or am I missing something about random number generations? Will the sequence be different from one computer to the next?

Reply:
Ah, I guess I missed the `srand` somewhere earlier, my bad :blush:

Reply:
`srand` is called inside `get_random_game_options` 😉